yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -66,6 +66,8 @@ private[spark] class Client(
   private val executorMemoryOverhead = args.executorMemoryOverhead // MB
   private val distCacheMgr = new ClientDistributedCacheManager()
   private val isClusterMode = args.isClusterMode
+  private val fireAndForget = isClusterMode &&
+    sparkConf.getBoolean("spark.yarn.cluster.quiet", false)
```

Comment:
I actually slightly prefer `spark.yarn.submit.waitForCompletion`. To me it's not ambiguous what it's waiting for. No strong feelings.